apis/condition_status_type.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apis
+
+import (
+	"reflect"
+	"sort"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ConditionType string
+
+const (
+	// ConditionReady specifies that the resource is ready.
+	ConditionReady ConditionType = "Ready"
```

Comment:
"Ready" isn't a meaningful term in every Knative project. `knative/build`'s condition type is `BuildSuccessful`. Should we remove this value from `knative/pkg` and copy it in `knative/serving` and `knative/eventing`?

Reply:
The majority of resources will have a "Ready", and those that do not don't have to use this. that controller can always create their own ConditionType strings. (or just send a string in.)

Reply:
Possibly "Ready" for build could mean a controller is watching the resource...

Reply:
Maybe, but that's not what `Ready` means for other knative resources, so it would probably be confusing. I'm fine with having `Ready` defined in this shared package, so long as it's clear it has no special meaning to the package except that it's a commonly used condition type across knative resources.

Reply:
What if you did use `Ready` and `Reason` was `BuildSuccessful`

Reply:
That would be a pretty big semantic shift from what our condition means today. Unlike a service or event stream, a build is run to completion and then is just "done" -- maybe it could be considered "ready" for some downstream thing to act on it, but I don't think we can make that statement about all builds.

Reply:
https://github.com/knative/serving/blob/master/docs/spec/errors.md#conditions defines two 'top-level "happy state" conditions':

* For long-running resources, this condition type should be `Ready`.
* For objects which run to completion, the condition type should be `Succeeded`.

Reply:
Great, if that's already part of our API spec, then maybe this package should just define both condition types, with comments pointing to the spec.

Reply:
I have added it